src/Components/test/E2ETest/Tests/JSRootComponentsTest.cs

```diff
@@ -200,6 +201,26 @@ public void CanSupplyCatchAllParameters()
                 });
         }
 
+        [Fact]
+        public void CanSupplyAndInvokeFunctionParameters()
+        {
+            var containerId = "root-container-1";
+
+            app.FindElement(By.Id("add-root-component")).Click();
+            app.FindElement(By.Id("set-function-params")).Click();
+            Browser.Equal($"Finished setting function parameters on component in {containerId}", () => app.FindElement(By.Id("message")).Text);
+
+            var container = Browser.FindElement(By.Id(containerId));
+
+            // Invoke callback without params.
+            container.FindElement(By.ClassName("js-callback")).Click();
+            Browser.Equal($"JavaScript button callback invoked", () => app.FindElement(By.Id("message")).Text);
+
+            // Invoke callback with params.
+            container.FindElement(By.ClassName("js-callback-with-params")).Click();
+            Browser.Equal($"JavaScript button callback received mouse event args (buttons=0)", () => app.FindElement(By.Id("message")).Text);
+        }
```

Comment:
Do you think it's worth having an E2E test showing that the eventcallback functions can be updated and/or nulled out, since there's a separate code path handling that case?

Reply:
Please feel free to merge whenever you're ready, however you choose to handle or not handle this suggestion. Up to you whether it adds enough value to warrant doing.

Reply:
I think it's worth it - I've already started to add a few more tests, just haven't finished them yet.